core/server.go

```diff
@@ -107,18 +105,6 @@ func WithRestManager(manager Manager) ServerOption {
 	return &restManagerOption{manager}
 }
 
-type loggerOption struct {
```

Comment:
Add the following function to support the `pkg` logger:
```
func WithPkgLogger(configs ...Config) ServerOption { ... }
```

Internally we only want to use the `pkg` logger, so whether `WithLogrusLogger` or `WithPkgLogger` or neither are called we will set a `pkg` logger against the context. I'll detail how these should be treated separately:

#### WithPkgLogger

Set a `pkg` logger against the context with the provided `configs`.

There's an interesting piece of code in the `NewRequestLogger` function that only performs a certain action if the `logrus` log level was high enough. The `logrus` log level was effectively being used to describe the verbosity of `sysl-go` logging (which although similar, are two different concepts).

We want to retain the idea of `sysl-go` logging verbosity with the `pkg` logger. We could add another `ServerOption` to indicate this but I'd rather leave it out of the interface for now and use the log level of the `logrus` logger (as previous) and the `verbose` level of the `pkg` logger to indicate this value.

The complication is that the `pkg` logger doesn't expose whether or not it is logging in a `verbose` manner, however you can check whether or not one of the `configs` is the verbose config, so this will have to do for now.

To communicate the `sysl-go` logging verbosity throughout the rest of `sysl-go`, a value should be added to the context in the `ServerParams` and pulled out when necessary.

#### WithLogrusLogger

Set a `pkg` logger against the context and use the `AddHooks` config value to forward `pkg` logger entries to the provided `logrus` logger.

Also add the `logrus` logger to the config to ensure it works in the same way as it did before.

Deprecate all exported methods that deal with the `logrus` logger directly.

#### No logger set

Treat this the same as if `WithPkgLogger` was called with no `configs`.

Reply:
Just want to make sure if we have the configs for `WithPkgLogger`, how do we pass in the configs if we still want backward compatible?